### ISSUE TITLE
Make BM_PlatformMessageResponseDartComplete less noisy

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#83360]
See https://flutter-engine-perf.skia.org/e/?begin=1621046942&end=1621960741&keys=X734bc6c23dcbc7982873405467ffe9e0&requestType=0&xbaroffset=22732.

Looks like it's around commit https://github.com/flutter/engine/pull/26391. I'd just approve it, because it looks like it's test-only, except that the PR says that the tests are "inexpensive", and the dashboard says otherwise.

**Comment by zanderso** [comment]
https://github.com/flutter/engine/pull/26391 was a partial revert of a change to the Cirrus presubmit checks, so is very unlikely to be it. One of the subsequent commits might be the cause, but I'm guessing this needs a bisect to identify.

/cc @GaryQian is the current sheriff.

**Comment by GaryQian** [comment]
Bisecting

**Comment by zanderso** [comment]
Looking at what the benchmark is doing, it is mostly measuring the time to copy a 3MB buffer and a thread synchronization. The small workload plus a thread synchronization will make the results noisy since we're not in control of the thread schedule. 

I tried to use a smaller buffer, hoping that the benchmark harness would do more runs to even out the variability from thread synchronization. I also tried putting a `for (int i = 0; i < 1000; i++) {}` under the `while (state.KeepRunning()) {}` to force that to happen to some extent, but neither approach seemed to help very much. The benchmark was mostly flat with occasional large dips and spikes. I think we'll have to accept this shift in the benchmark for now, but I'm going to repurpose this issue to be about investigating making the benchmark less noisy.
